src/core/services/compiler/compiler.spec.js

```diff
@@ -164,12 +164,24 @@ describe('$mdCompiler service', function() {
 
   });
 
-  [
+  var bindingStatesToTest;
+  if (angular.version.major === 1 && angular.version.minor >= 7) {
```

Comment:
That's not 100% necessary but a test checking the `respectPreAssignBindingsEnabled` setter throws in 1.7 would be awesome.

Reply:
I gave it a shot, but I'm not sure how to test an exception in a module `config` function and calling it outside of that isn't allowed (not a function).

Reply:
This may be doable, the main problem is it's not possible to call `module` (which is used in tests to access the config phase) after `inject` was called at least once. You'd have to move `inject` from `beforeEach` of `it` blocks so that at the beginning of a new `it` you can still call:
```js
module(function ($mdCompilerProvider) {
  expect(function () {
    $compileProvider.preAssignBindingsEnabled(false);
  }).toThrow();
});
```
I'm not 100% sure if an error thrown during the config phase doesn't break the test infrastructure but maybe it doesn't.

Reply:
You can test that it throws with something like:

```js
it('should throw ...', () => {
  module('material.core', $mdCompilerProvider =>
    $mdCompilerProvider.respectPreAssignBindingsEnabled(false));
  expect(inject).toThrowError(regexThatMatchesTheErrorMessage);
```